src/commands/tag/set-missing.command.ts

```diff
@@ -0,0 +1,78 @@
+import { clone, Command, Input } from "../../deps.ts";
+import { CmdGlobalOptions } from "../cmd-options.ts";
+import { setupLogger } from "../../logger.ts";
+import { verifyCliAvailability } from "../../init.ts";
+import { loadConfig } from "../../config/config.model.ts";
+import { MeshAdapterFactory } from "../../mesh/mesh-adapter.factory.ts";
+import { MeshAdapter } from "../../mesh/mesh-adapter.ts";
+import { MeshTenant } from "../../mesh/mesh-tenant.model.ts";
+import { MeshInvalidTagValueError } from "../../errors.ts";
+
+export const setMissingTags = new Command()
+  .description(
+    "Fix all tenants missing the given tag. Collie will ask you per tenant what the value should be and writes the change to the cloud platform(s).",
+  )
+  .action(setMissingTagsAction);
+
+async function setMissingTagsAction(options: CmdGlobalOptions, tagKey: string) {
+  setupLogger(options);
+  await verifyCliAvailability();
+
+  const config = loadConfig();
+  const meshAdapterFactory = new MeshAdapterFactory(config);
+  const meshAdapter = meshAdapterFactory.buildMeshAdapter(options);
+
+  const allTenants = await meshAdapter.getMeshTenants();
+
+  const tenantsMissingTag = allTenants.filter((x) =>
+    !x.tags.find((y) => y.tagName === tagKey)
+  );
+
+  console.log(
+    `We have identified ${tenantsMissingTag.length} tenants without the tag "${tagKey}". You'll be able to set the tag values for these tenants now.`,
+  );
+
+  let count = 1;
+  for (const tenant of tenantsMissingTag) {
+    await askAndSetTag(
+      tagKey,
+      tenant,
+      meshAdapter,
+      count,
+      tenantsMissingTag.length,
+    );
+    count++;
+  }
+}
+
+async function askAndSetTag(
+  tagKey: string,
+  tenant: MeshTenant,
+  meshAdapter: MeshAdapter,
+  count: number,
+  totalCount: number,
+) {
+  const prefix = `(${count}/${totalCount})`;
+  const tagValue: string = await Input.prompt({
+    message:
+      `${prefix} For the tenant ${tenant.platformTenantName} (${tenant.platformTenantId}) on ${tenant.platform}, what tag value do you want to set for "${tagKey}"? (Leave empty to skip)`,
+  });
+
+  if (tagValue !== "") {
+    // Skip this tenant if the user does not give any value.
+    const originalTenant = clone(tenant);
+    tenant.tags.push({ tagName: tagKey, tagValues: [tagValue] });
+    try {
+      await meshAdapter.updateMeshTenant(tenant, originalTenant);
+    } catch (e) {
+      if (e instanceof MeshInvalidTagValueError) {
+        console.log(
+          `You have entered an invalid tag value (${tagValue}) which is not supported by the platform. Please try again with a different value.`,
+        );
+        await askAndSetTag(tagKey, tenant, meshAdapter, count, totalCount);
```

Comment:
d: this will basically only work once and if the user inputs a wrong tag the second time it will bail out. Might be desirable but I think the change in behavior is weird. Maybe put it into a loop until the user got it right? But I am also fine with this behavior. No strong oppinion.

Reply:
It's recursive so it will keep trying until the user enters a valid tag. It works as expected:
![image](https://user-images.githubusercontent.com/14248574/135999221-0db531a2-c2e3-477d-8818-8b8d4255d94b.png)